src/libponyc/ast/ast.c

```diff
@@ -1221,6 +1221,85 @@ void ast_free_unattached(ast_t* ast)
     ast_free(ast);
 }
 
+static void get_next_token_pos(ast_t* ast, size_t* line, size_t* pos)
+{
+  assert(ast != NULL);
+  
+  while (ast->child != NULL)
+  {
+    ast = ast->child;
+    while (ast->sibling != NULL)
+      ast = ast->sibling;
+  }
+
+  *line = token_line_number(ast->t);
+  *pos = token_line_position(ast->t);
+
+  token_id id = ast_id(ast);
+  if (id == TK_STRING || id == TK_ID)
+    *pos += token_string_len(ast->t);
```

Comment:
Note that this approach would also help to avoid weird formatting issues, like line breaks or comments in the middle of the parameter list.